paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -209,9 +209,34 @@ internal class CustomerSheetViewModel @Inject constructor(
         // TODO (jameswoo) handle onFormValuesChanged
     }
 
-    @Suppress("UNUSED_PARAMETER")
     private fun onItemRemoved(paymentMethod: PaymentMethod) {
-        TODO()
+        viewModelScope.launch {
+            paymentMethod.id?.let { paymentMethodId ->
+                customerAdapter.detachPaymentMethod(paymentMethodId = paymentMethodId)
+                    .onFailure { throwable ->
+                        updateViewState<CustomerSheetViewState.SelectPaymentMethod> {
+                            it.copy(
+                                errorMessage = throwable.message,
```

Comment:
I don't think we should pass throwable messages through to user visible error messages.

Reply:
This is coming from Stripe API

Reply:
Maybe in some cases, but what about network timeout, or connections issues, will this always be a user friendly error message?

Reply:
It is handled by our networking API layer

Reply:
Using generic error message for now. Will get feedback on this later.